### PR TITLE
Implement common quantification

### PR DESCRIPTION
**Pull request by nictru**
Currently, circRNA expression values are obtained by taking the highest value of all detection tools that found the circRNA. This can be biased if certain detection tools deliver higher values than others. With this PR, the circRNA locations which are obtained by the detection tools are used as input for a single quantification using `psirc-quant` which is based on kallisto. It also considers the linear transcriptome, so that an unbiased analysis of circRNAs together with the linear transcripts becomes possible.

Requires #104 to be merged first.

### CODE REVIEW

**Comment by rreggiar** [comment]
I like this and came to the same conclusion in conversation with other circRNA folks. We may consider having multiple "aggregate" quantifiers, but I do like psirc's approach the most. The community may want something else either in-place or in-parallel.

**Comment by nictru** [comment]
Great - the new `quantification` subworkflow should make it easy to add other tools with comparable function. Psirc fulfills the needs in my project for now, but if new wishes come up I am sure we can add them. If something already comes to your mind, feel free to point it out here. 

Also the next thing I plan to do in this context is to use the outputs of the `quantification` subworkflow in the `differential_expression` subworkflow. Currently the counts from the tool aggregation in the `circrna_detection` subworkflow is used along with a separate quantification for genes (`stringtie`). I can imagine that at least the `circtest` will work better when based on the common quantification, but most likely DESeq too.

**Comment by nictru** [comment]
I finished the base functionality now. We could easily add `salmon` as an alternative quantifier later on. I also added a parameter for `bootstrap_samples` which will later on allow us to consider inferential uncertainty during differential expression. This can be based on the `SummarizedExperiment` that is generated by the `tximeta` process.

File: modules/local/count_matrix/combined/main.nf [hunk 1]
```diff
@@ -1,37 +1,28 @@
 process COUNTS_COMBINED {
-    label 'process_low'
+    label "process_single"
 
-    conda "conda-forge::r-base=3.6.3 conda-forge::python=2.7.15 conda-forge::r-argparser=0.6 conda-forge::r-dplyr=1.0.5"
+    conda "conda-forge::mulled-v2-2076f4a3fb468a04063c9e6b7747a630abb457f6==fccb0c41a243c639e11dd1be7b74f563e624fcca-0"
```
Comment (maxulysse) [inline modules/local/count_matrix/combined/main.nf]: I'm not sure this is working

**Review by maxulysse — COMMENTED** [review]
Looks clear enough, a couple of comments about handling conda with some local modules

File: modules/local/ciriquant/ciriquant/main.nf
Comment (maxulysse) [inline modules/local/ciriquant/ciriquant/main.nf:1]: `quay.io` is not needed in the URI definition in this container

File: modules/local/count_matrix/merge_tools/main.nf [hunk 2]
```diff
@@ -1,45 +1,31 @@
 process MERGE_TOOLS {
     tag "$meta.id"
-    label 'process_low'
+    label "process_single"
 
-    conda "conda-forge::r-base=3.6.3 conda-forge::python=2.7.15 conda-forge::r-argparser=0.6 conda-forge::r-dplyr=1.0.5"
+    conda "conda-forge::mulled-v2-2076f4a3fb468a04063c9e6b7747a630abb457f6==fccb0c41a243c639e11dd1be7b74f563e624fcca-0"
```
Comment (maxulysse) [inline modules/local/count_matrix/merge_tools/main.nf]: Same issue here

Comment (nictru) [reply modules/local/count_matrix/merge_tools/main.nf:5]: I did a global search for this and removed all of them, so no need to point out each individually :)

File: modules/local/ciriquant/ciriquant/main.nf
Comment (nictru) [reply modules/local/ciriquant/ciriquant/main.nf:1]: I did a global search for this and removed all of them, there were some more occurrences

File: modules/local/count_matrix/merge_tools/main.nf (hunk 2 above)
Comment (nictru) [reply modules/local/count_matrix/merge_tools/main.nf]: I will check how many of these exist

Comment (nictru) [reply modules/local/count_matrix/merge_tools/main.nf]: Global search said this is the only remaining occurrence of `conda-forge::mulled` in the repo

File: modules/local/ciriquant/ciriquant/main.nf [hunk 3]
```diff
@@ -52,7 +52,7 @@ process CIRIQUANT {
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
         bwa: \$(echo \$(bwa 2>&1) | sed 's/^.*Version: //; s/Contact:.*\$//')
-        ciriquant : \$(echo \$(CIRIquant --version 2>&1) | sed 's/CIRIquant //g' )
+        ciriquant: \$(echo \$(CIRIquant --version 2>&1) | sed 's/CIRIquant //g' )
         samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
         stringtie: \$(stringtie --version 2>&1)
         hisat2: $VERSION
```
Comment (maxulysse) [inline modules/local/ciriquant/ciriquant/main.nf:58]: ok, so ciriquant has bwa, samtools, stringtie and hisat2 as well, that's interesting

File: modules/local/psirc/index/main.nf [hunk 4]
```diff
@@ -0,0 +1,24 @@
+process PSIRC_INDEX {
+    tag "${meta.id}"
+    label 'process_high'
+
+    container "registry.hub.docker.com/bigdatainbiomedicine/psirc"
```
Comment (maxulysse) [inline modules/local/psirc/index/main.nf:5]: not available in conda I assume.
Is there any Dockerfile available for this tool?

**Review by maxulysse — APPROVED** [review]

File: modules/local/ciriquant/ciriquant/main.nf (hunk 3 above)
Comment (nictru) [reply modules/local/ciriquant/ciriquant/main.nf:58]: It actually requires all of these as [described here](https://ciri-cookbook.readthedocs.io/en/latest/CIRIquant_1_installation.html#prerequisites)

File: modules/local/psirc/index/main.nf (hunk 4 above)
Comment (nictru) [reply modules/local/psirc/index/main.nf:5]: I will add finding a cleaner solution to my todo list

Comment (nictru) [reply modules/local/psirc/index/main.nf:5]: Added #115

**Pull request merged**